### ISSUE TITLE
Pod build failure.

### ISSUE THREAD

**Issue by AndrewLeeCHCH** [heroiclabs/nakama-swift#8]
When I finished pod install, the build will fail because of a swift compiler error listed as below.
![image](https://user-images.githubusercontent.com/7780415/53837497-88b06400-3f47-11e9-8751-44a9c29116ba.png)

By the way, I have fixed issues in PromiseKit due to swift version 

My swift toolchain: 
Apple Swift version 4.2.1 (swiftlang-1000.11.42 clang-1000.11.45.1)
Target: x86_64-apple-darwin18.2.0

My Xcode version: 
10.1

**Comment by AndrewLeeCHCH** [comment]
For this build issue, I do find a quick fix like this.
![image](https://user-images.githubusercontent.com/7780415/53838195-48ea7c00-3f49-11e9-8718-be738723668d.png)

After that, I can build the project and run it. 
However, when I try to register/login with my client, it crashed.
![image](https://user-images.githubusercontent.com/7780415/53838315-91099e80-3f49-11e9-9903-7032b0093628.png)

I looked into SwiftProtobuf.BinaryDecodingError. The error is listed below.
![image](https://user-images.githubusercontent.com/7780415/53838375-c0201000-3f49-11e9-8b0b-101315c2920e.png)

To make sure the client setting is correct, I used the same client setting in Java and it worked perfectly.

**Comment by mofirouz** [comment]
@AndrewLeeCHCH Could you clarify what version of the Nakama Swift Client you are using? Is it a tagged pod, or a commit from master?

We do have plans on improve/complete the Swift client code, as well as update to latest Swift version but haven't had time yet.

**Comment by AndrewLeeCHCH** [comment]
@mofirouz I use tagged pod.
Pod file is 
pod 'Nakama', '~> 0.3'

Podfile.lock is 
PODS:
  - Nakama (0.3.0):
    - PromiseKit (~> 4)
    - Starscream (~> 2)
    - SwiftProtobuf (~> 1)
  - PromiseKit (4.5.2):
    - PromiseKit/Foundation (= 4.5.2)
    - PromiseKit/QuartzCore (= 4.5.2)
    - PromiseKit/UIKit (= 4.5.2)
  - PromiseKit/CorePromise (4.5.2)
  - PromiseKit/Foundation (4.5.2):
    - PromiseKit/CorePromise
  - PromiseKit/QuartzCore (4.5.2):
    - PromiseKit/CorePromise
  - PromiseKit/UIKit (4.5.2):
    - PromiseKit/CorePromise
  - Starscream (2.1.1)
  - SwiftProtobuf (1.3.1)

DEPENDENCIES:
  - Nakama (~> 0.3)

SPEC REPOS:
  https://github.com/cocoapods/specs.git:
    - Nakama
    - PromiseKit
    - Starscream
    - SwiftProtobuf

SPEC CHECKSUMS:
  Nakama: fe88fb2334fc1b22e3b3b233caa5eecd4cd08adc
  PromiseKit: 743e497a5f505a470d3bbbf4ce0663c1268af0a4
  Starscream: 142bd8ef24592d985daee9fa48c936070b85b15f
  SwiftProtobuf: 84cd6b8f4fd2fc3d9fbd80a0aaf2e957d52824ec

PODFILE CHECKSUM: 56308fdc0b5c1d9847016c525d82a6f2bfabd5c0

COCOAPODS: 1.5.3

**Comment by mofirouz** [comment]
Interesting - so you are planning to use Nakama 1.4.x?

**Comment by AndrewLeeCHCH** [comment]
To be honest, I am not quite sure about which version I should use. The basic need is login, connect, update info and create match. Moreover, I would like to use Nakama Client which dependency of PromiseKit is ~> 6

**Comment by mofirouz** [comment]
@AndrewLeeCHCH One of the community members (@amradawi) updated the Swift client on master to support Nakama 2.0 - the implementation is incomplete and in-flux at the moment. You could and use that, which I believe uses PMK 6.0 - but I've not tested this and there is no pod for it either atm.

**Comment by AndrewLeeCHCH** [comment]
Cool, I will definitely give it a try. Thx

**Comment by AndrewLeeCHCH** [comment]
@mofirouz 
![image](https://user-images.githubusercontent.com/7780415/53917394-3fc8e000-4019-11e9-86fa-cd4b5a020496.png)

![image](https://user-images.githubusercontent.com/7780415/53917260-f9738100-4018-11e9-92c2-f285cd1167ba.png)
A quick question about Nakama Client in master branch. I have updated pod file to get Nakama2.0. Unfortunately, I found Nakama Folder is empty. Did I make any stupid mistake which lead to loss of Nakama Client itself?
